### PR TITLE
Add new provider google_organization_iam_audit_config.

### PR DESCRIPTION
**Pull request by xingao267**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_organization_iam_audit_config`
```

### CODE REVIEW

**Comment by xingao267** [comment]
Hi @danawillow, could you help review this PR?

And what's the expected timeline for this new resource to be available in terraform-provider-google-beta release and terraform-provider-google release?

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    log_type = "DATA_READ"%s
+  }
+}
+`, org, service, memberStr)
+}
+
+func testAccOrganizationAssociateAuditConfigLogType(org, service, logType string) string {
+	return fmt.Sprintf(`
+resource "google_organization_iam_audit_config" "acceptance" {
+	org_id = "%s"
```
Comment (slevenick) [inline third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Spacing consistency issues

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	var memberStr string
+	if len(members) > 0 {
+		for pos, member := range members {
+			members[pos] = "\"" + member + "\","
+		}
+		memberStr = "\n    exempted_members = [" + strings.Join(members, "\n") + "\n    ]"
+	}
+	return fmt.Sprintf(`
+resource "google_organization_iam_audit_config" "acceptance" {
+	org_id = "%s"
```
Comment (slevenick) [inline third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Spacing

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    ]
+  }
+}
+`, org, service)
+}
+
+func testAccOrganizationAssociateAuditConfigDropMemberFromBasic(org, service string) string {
+	return fmt.Sprintf(`
+resource "google_organization_iam_audit_config" "acceptance" {
+	org_id = "%s"
```
Comment (slevenick) [inline third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Spacing

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    log_type = "DATA_WRITE"
+  }
+}
+`, org, service, org, service2)
+}
+
+func testAccOrganizationAssociateAuditConfigUpdated(org, service string) string {
+	return fmt.Sprintf(`
+resource "google_organization_iam_audit_config" "acceptance" {
+	org_id = "%s"
```
Comment (slevenick) [inline third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Spacing

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    log_type = "DATA_READ"
+    exempted_members = [
+		  "user:paddy@hashicorp.com",
+      "user:paddy@carvers.co",
+    ]
+  }
+}
+
+resource "google_organization_iam_audit_config" "multiple" {
+	org_id = "%s"
```
Comment (slevenick) [inline third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Spacing

Comment (xingao267) [reply third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Done.

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go (hunk 4 above)
Comment (xingao267) [reply third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Done.

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go (hunk 3 above)
Comment (xingao267) [reply third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Done.

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go (hunk 2 above)
Comment (xingao267) [reply third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Done.

File: third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go (hunk 1 above)
Comment (xingao267) [reply third_party/terraform/tests/resource_google_organization_iam_audit_config_test.go]: Done.

**Comment by xingao267** [comment]
@slevenick I was able to run the tests locally with an org that I have permission to modify.

I'm guessing the organization IAM policy tests pass because they are skipped? I noticed that the tests are skipped if GOOGLE_ORG and GOOGLE_ORG_2 (required for iam org policy tests) are not set properly.

**Comment by slevenick** [comment]
Can you add the ability to check this environment variable and skip the tests similar to:
https://github.com/terraform-providers/terraform-provider-google/blob/03cf878c8f8c6367354d3d0657a102a019172d28/google/resource_google_organization_iam_test.go#L23

First of all, in our CI system these tests are going to be run in parallel which will cause them to fail due to them all operating on the same organization.

Second, we tend to skip organization IAM tests because they potentially impact other tests going on. Even though these tests are on the IAM audit config, they require operating on the org IAM policy itself, so I would prefer to skip these during normal runs.

I ran these locally and they seem fine

**Comment by xingao267** [comment]
@slevenick  Done.

Just out of curiosity, I didn't add the `t.Parallel()` as in other tests, but they still get run in parallel in the CI?

**Comment by slevenick** [comment]
> @slevenick Done.
> 
> Just out of curiosity, I didn't add the `t.Parallel()` as in other tests, but they still get run in parallel in the CI?

Yeah, it's not ideal but there isn't currently a way to specify that a series of tests should not be run in parallel with each other except for combining them into a single test.

**Comment by xingao267** [comment]
@slevenick Thanks for the review!

**Pull request merged**